src/cdk/testing/protractor/protractor-element.ts

```diff
@@ -198,4 +198,24 @@ export class ProtractorElement implements TestElement {
   async isFocused(): Promise<boolean> {
     return this.element.equals(browser.driver.switchTo().activeElement());
   }
+
+  async dispatchEvent(name: string, data?: Record<string | number, any>): Promise<void> {
+    return browser.executeScript(_dispatchEvent, name, data ? JSON.stringify(data) : null,
```

Comment:
`JSON.stringify(data)` seems problematic for setting things like `target` which is quite a common use case for events. Is there anything we can do about this?

Reply:
The `target` gets set automatically by the browser when it dispatches the event. We could narrow down the type to only include primitives?

Reply:
According to the [`executeScript` docs](https://www.protractortest.org/#/api?view=webdriver.WebDriver.prototype.executeScript)

> Arguments may be a boolean, number, string, or WebElement. Arrays and objects may also be used as script arguments as long as each item adheres to the types previously mentioned.

Given that I think we should tighten up the type for data and _not_ use `JSON.stringify` on it, as that will not handle `WebElement` correctly

Reply:
Is a `WebElement` one-to-one representation of a native DOM `Node` though? If it's not, I'm not sure that it would make sense to assign to an event anyway. My thinking behind the `data` was primarily to be able to stub out custom events.